Rubberduck.RibbonDispatcher/Concrete/AbstractDispatcher.cs

```diff
@@ -0,0 +1,147 @@
+            => Controls(control?.Id)?.Size ?? RdControlSize.rdLarge;
+
+        /// <inheritdoc/>
+        public object GetImage(IRibbonControl control)
+            => Imageables(control?.Id)?.Image ?? "MacroSecurity";
+        /// <inheritdoc/>
+        public bool   GetShowImage(IRibbonControl control)
+            => Imageables(control?.Id)?.ShowImage ?? true;
+        /// <inheritdoc/>
+        public bool   GetShowLabel(IRibbonControl control)
+            => Imageables(control?.Id)?.ShowLabel ?? true;
+
+        /// <inheritdoc/>
+        public bool   GetPressed(IRibbonControl control)
+            => Toggles(control?.Id)?.IsPressed ?? false;
+        /// <inheritdoc/>
+        public void   OnActionToggle(IRibbonControl control, bool pressed)
+            => Toggles(control?.Id)?.OnActionToggle(pressed);
+
+        /// <inheritdoc/>
+        public void   OnAction(IRibbonControl control) => Actions(control?.Id)?.OnAction();
+
+        /// <inheritdoc/>
+        public string GetSelectedItemId(IRibbonControl control)
+            => DropDowns(control?.Id)?.SelectedItemId;
+        /// <inheritdoc/>
+        public int    GetSelectedItemIndex(IRibbonControl control)
+            => DropDowns(control?.Id)?.SelectedItemIndex ?? 0;
+        /// <inheritdoc/>
+        public void   OnActionDropDown(IRibbonControl control, string selectedId, int selectedIndex)
+            => DropDowns(control?.Id)?.OnActionDropDown(selectedId, selectedIndex);
+ 
+        /// <inheritdoc/>
+        public int    GetItemCount(IRibbonControl control)
+            => DropDowns(control?.Id)?.ItemCount ?? 0;
+        /// <inheritdoc/>
+        public string GetItemId(IRibbonControl control, int index)
+            => DropDowns(control?.Id)?.ItemId(index) ?? "";
+        /// <inheritdoc/>
+        public string GetItemLabel(IRibbonControl control, int index)
+            => DropDowns(control?.Id)?.ItemLabel(index) ?? Unknown(control);
+        /// <inheritdoc/>
+        public string GetItemScreenTip(IRibbonControl control, int index)
+            => DropDowns(control?.Id)?.ItemScreenTip(index) ?? Unknown(control);
+        /// <inheritdoc/>
+        public string GetItemSuperTip(IRibbonControl control, int index)
+            => DropDowns(control?.Id)?.ItemSuperTip(index) ?? Unknown(control);
+
+        /// <inheritdoc/>
+        public object GetItemImage(IRibbonControl control, int index)
+            => DropDowns(control?.Id)?.ItemImage(index) ?? "MacroSecurity";
+        /// <inheritdoc/>
+        public bool   GetItemShowImage(IRibbonControl control, int index)
+            => DropDowns(control?.Id)?.ItemShowImage(index) ?? true;
+        /// <inheritdoc/>
+        public bool   GetItemShowLabel(IRibbonControl control, int index)
+            => DropDowns(control?.Id)?.ItemShowLabel(index) ?? true;
+
+        //private ISelectableItem DropDownItem(IRibbonControl control, int index)
+        //    => DropDowns(control?.Id)[index];
```

Comment:
commented out code... why? Explain in a comment or remove, please 😄

Reply:
Started on another round of implementation, then changed my mind. This will be used in the next round, when I clean up DropDowns in preparation for implementing ComboBoxes.